### ISSUE TITLE
Segmentation fault (core dumped) in toolbox using the COPR version

### ISSUE THREAD

**Issue by Vilianexe** [Genymobile/scrcpy#3843]
- [x] I have read the [FAQ](https://github.com/Genymobile/scrcpy/blob/master/FAQ.md).
 - [x] I have searched in existing [issues](https://github.com/Genymobile/scrcpy/issues).

**Environment**
 - OS: Fedora Kinoite 38
 - scrcpy version: scrcpy 2.0
 - installation method: COPR
 - device model: Samsung galaxy M21s
 - Android version: 12

**Describe the bug**
scrcpy just show Segmentation fault in nothing more(even when using --verbosity=verbose) 

```
⬢[vdaros@toolbox ~]$ scrcpy --verbosity=verbose
scrcpy 2.0 <https://github.com/Genymobile/scrcpy>
Segmentation fault (core dumped)
⬢[vdaros@toolbox ~]$ 
```

Please do not post screenshots of your terminal, just post the content as text instead.

**Comment by rom1v** [comment]
Thank you for your report.

Please try installing the latest version manually: https://github.com/Genymobile/scrcpy/blob/master/doc/linux.md#latest-version (remove the copr package first)

Does it still segfault?

**Comment by Vilianexe** [comment]
> Does it still segfault?

yep, i even created other toolbox to guaranteed that the COPR was completed removed, and same error
```shell
scrcpy --verbosity=verbose
scrcpy 2.0 <https://github.com/Genymobile/scrcpy>
Segmentation fault (core dumped)

```

**Comment by rom1v** [comment]
OK, then could you [build manually](https://github.com/Genymobile/scrcpy/blob/master/doc/build.md#option-2-use-prebuilt-server)), but in debug mode with ASAN enabled:

```bash
# build
meson setup d -Db_sanitize=address -Dprebuilt_server=/path/to/scrcpy-server
ninja -Cd

# run
./run d
```

**Comment by rom1v** [comment]
Did you close because the is solved?

> OK, then could you please [build manually](https://github.com/Genymobile/scrcpy/blob/master/doc/build.md#option-2-use-prebuilt-server), but in debug mode with ASAN enabled:

What was reported by ASAN?

**Comment by akippnn** [comment]
> Did you close because the is solved?
> 
> > OK, then could you please [build manually](https://github.com/Genymobile/scrcpy/blob/master/doc/build.md#option-2-use-prebuilt-server), but in debug mode with ASAN enabled:
> 
> What was reported by ASAN?

I have a similar setup as above ([`bluefin-dx`](https://github.com/ublue-os/bluefin); based on Fedora Silverblue 39) and did get the same segmentation fault error with the COPR version of `scrcpy` when installing it first try.

So I followed the steps of building it manually with the prebuilt server. `./run d` works, but now so does `scrcpy`. I tried creating a new toolbox container and followed the same steps and suddenly `scrcpy` works OOTB.

Not sure what's going on. Here's the ASAN stderr output: https://gist.github.com/akippnn/dc3fabf58aa728f1024bef9a3ddb0486
